app/models/service.rb

```diff
@@ -184,6 +223,20 @@ def queue_group_action(action, group_idx, direction, deliver_delay)
     true
   end
 
+  def queue_power_calculation(extend_delay = 0)
+    calculate_power = {
+      :class_name  => self.class.name,
+      :instance_id => id,
+      :method_name => "calculate_power_state",
+      :role        => "ems_operations",
+      :task_id     => "#{self.class.name.underscore}_#{id}",
```

Comment:
Calculating the power_state does not rely on the provider so you do not have to queue this to the `ems_operations` role (you can leave it empty) and you should not need the `task_id` either.